src/feature/dircache/dircache_stub.c

```diff
@@ -0,0 +1,152 @@
+/* Copyright (c) 2001-2004, Roger Dingledine.
+ * Copyright (c) 2004-2006, Roger Dingledine, Nick Mathewson.
+ * Copyright (c) 2007-2020, The Tor Project, Inc. */
+/* See LICENSE for licensing information */
+
+/**
+ * @file dircache_stub.c
+ * @brief Stub declarations for use when dircache module is disabled.
+ **/
+
+#include "core/or/or.h"
+#include "feature/dircache/consdiffmgr.h"
+#include "feature/dircache/dircache.h"
+#include "feature/dircache/dirserv.h"
+#include "feature/dircommon/dir_connection_st.h"
+
+int
+directory_handle_command(dir_connection_t *conn)
+{
+  (void) conn;
+  tor_assert_nonfatal_unreached_once();
+  return -1;
+}
+
+int
+connection_dirserv_flushed_some(dir_connection_t *conn)
+{
+  (void) conn;
+  tor_assert_nonfatal_unreached_once();
+  return -1;
+}
+
+int
+directory_fetches_from_authorities(const or_options_t *options)
+{
+  (void) options;
+  return 0;
+}
+
+int
+directory_fetches_dir_info_early(const or_options_t *options)
+{
+  (void) options;
+  return 0;
+}
+
+int
+directory_fetches_dir_info_later(const or_options_t *options)
+{
+  (void) options;
+  return 0;
+}
+
+int
+directory_caches_unknown_auth_certs(const or_options_t *options)
+{
+  (void) options;
+  return 0;
+}
+
+int
+directory_caches_dir_info(const or_options_t *options)
+{
+  (void) options;
+  return 0;
+}
+
+int
+directory_permits_begindir_requests(const or_options_t *options)
+{
+  (void) options;
+  return 0;
+}
+
+int
+directory_too_idle_to_fetch_descriptors(const or_options_t *options,
```

Comment:
Despite its name, this is actually a client function:
* it depends on `rep_hist_circbuilding_dormant()`, which is a client function, and
* it depends on `FetchUselessDescriptors`, which is a general option in the man page, but probably belongs in an "Advanced Options" section, like `FetchDirInfoEarly`.

Reply:
(Unlike `FetchDirInfoEarly`, disabling the dircache module doesn't actually disable `FetchUselessDescriptors`, but I wouldn't mind if it did.)

Reply:
But if we decide to disable `FetchUselessDescriptors` when dircache is disabled, let's do it in a separate ticket.